From: Jan Kara <jack@suse.cz>
Date: Fri, 4 Nov 2016 18:08:14 +0100
Subject: ext2: Use clean_bdev_aliases() instead of iteration
References: bsc#1020989,FATE#322379
Git-commit: 69a9bea146b185be8ec50e80eaecd8e487e689f8
Patch-Mainline: v4.10-rc1

Use clean_bdev_aliases() instead of iterating through blocks one by one.

Signed-off-by: Jan Kara <jack@suse.cz>
Signed-off-by: Jens Axboe <axboe@fb.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 fs/ext2/inode.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/fs/ext2/inode.c b/fs/ext2/inode.c
index ccae181..16c9b38 100644
--- a/fs/ext2/inode.c
+++ b/fs/ext2/inode.c
@@ -734,6 +734,13 @@ static int ext2_get_blocks(struct inode *inode,
 
 	if (IS_DAX(inode)) {
 		/*
+		 * We must unmap blocks before zeroing so that writeback cannot
+		 * overwrite zeros with stale data from block device page cache.
+		 */
+		clean_bdev_aliases(inode->i_sb->s_bdev,
+				   le32_to_cpu(chain[depth-1].key),
+				   count);
+		/*
 		 * block must be initialised before we put it in the tree
 		 * so that it's not found by another thread before it's
 		 * initialised
-- 
1.8.5.6

